### PR TITLE
feat(sdl): can set credentials

### PR DESCRIPTION
**Pull request by jzsfkzm**
One can select an image repository and enter credentials after this. Available in the SDL Builder, but not in the deployment UI.

ref #467

### CODE REVIEW

**Review by baktun14 — COMMENTED** [review]
The credentials for private repos are optional, since not all repos are private. The way you implemented it is great, we just need to add a flag [like "hasGpu"](https://github.com/akash-network/console/blob/main/apps/deploy-web/src/types/sdlBuilder.ts#L80) and conditionally add the validation. So a simple checkbox like the `hasGpu` to toggle the form controls should do it.

Is there a reason behind the `canChangeCredentials`? We would like to have it in all the SDL builder instances ([SDL Builder page](https://console.akash.network/sdl-builder) and [new deployment page](https://console.akash.network/new-deployment?step=edit-deployment)).

File: apps/deploy-web/src/utils/sdl/data.ts [hunk 1]
```diff
@@ -17,10 +17,17 @@ export const defaultHttpOptions = {
...
+  host: 'docker.io',
+  username: '',
+  password: '',
+};
+
 export const defaultService: ServiceType = {
   id: nanoid(),
   title: "service-1",
   image: "",
+  credentials: defaultCredentials,
```
Comment (baktun14) [inline apps/deploy-web/src/utils/sdl/data.ts]: This is throwing and error in the build as the `username` and `password` are required.

**Comment by jzsfkzm** [comment]
> The credentials for private repos are optional, since not all repos are private. The way you implemented it is great, we just need to add a flag [like "hasGpu"](https://github.com/akash-network/console/blob/main/apps/deploy-web/src/types/sdlBuilder.ts#L80) and conditionally add the validation. So a simple checkbox like the `hasGpu` to toggle the form controls should do it.
> 
> Is there a reason behind the `canChangeCredentials`? We would like to have it in all the SDL builder instances ([SDL Builder page](https://console.akash.network/sdl-builder) and [new deployment page](https://console.akash.network/new-deployment?step=edit-deployment)).

I added the flag `canChangeCredentials` because I figured it is not needed for templates on the Deploy page. Now I see it is very much needed there too, so removed. Now it is all the same for Deploy and SDL Builder.
Added a checkbox with label "Private" on top the box, revealing credential fields only when you click it.

**Review by baktun14 — APPROVED** [review]
I just tested it locally and it works great, great work!

**Pull request merged**